ui/app/src/components/DialogHeader/DialogHeader.tsx

```diff
@@ -211,14 +212,27 @@ export function DialogHeader(props: DialogHeaderProps) {
         )}
       </section>
       {(subtitle || children) && (
-        <section className={cx(classes.subtitleWrapper, props.classes?.subtitleWrapper)}>
+        <Box
+          component="section"
+          className={cx(classes.subtitleWrapper, props.classes?.subtitleWrapper)}
+          sx={{ maxWidth: '100%' }}
+        >
           {subtitle && (
-            <Typography className={classes.subtitle} {...subtitleTypographyProps}>
+            <Typography
+              className={classes.subtitle}
+              {...subtitleTypographyProps}
+              sx={{
+                overflow: 'hidden',
+                textOverflow: 'ellipsis',
+                whiteSpace: 'nowrap'
+              }}
+              title={typeof subtitle === 'string' ? subtitle : null}
```

Comment:
This will affect nearly all dialogs. Move it to where you're needing it. Not good particularly on the verge of a release.

Add the sxs prop to this component so it can be done through sx and not just through classes.